# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/components.conf
# Commit: 79d687137fdf
# Full Hash: 79d687137fdf26d60122e947ad7198521e34bbf9
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2022-09-03 09:32:11
# Description:
#   Bug 1788856 - initialize NSS as needed in nsNSSCertificate r=jschanck
#   
#   Previously, instantiating an nsIX509Cert (implemented by nsNSSCertificate)
#   would cause NSS to be initialized. However, if 'new nsNSSCertificate()' was
#   called directly (rather than going through XPCOM), NSS would not be
# ==============================================================================

diff -r b14b975aa6d7 -r 79d687137fdf security/manager/ssl/components.conf
--- a/security/manager/ssl/components.conf	Fri Sep 02 22:15:58 2022 +0000
+++ b/security/manager/ssl/components.conf	Fri Sep 02 22:26:37 2022 +0000
@@ -51,7 +51,7 @@
         'cid': '{660a3226-915c-4ffb-bb20-8985a632df05}',
         'contract_ids': [],
         'type': 'nsNSSCertificate',
-        'legacy_constructor': 'mozilla::psm::NSSConstructor<nsNSSCertificate>',
+        'headers': ['nsNSSCertificate.h'],
     },
     {
         'cid': '{fb0bbc5c-452e-4783-b32c-80124693d871}',